src/main/java/org/jitsi/jicofo/AbstractChannelAllocator.java

```diff
@@ -199,6 +208,9 @@ private void doRun()
         catch (OperationFailedException e)
         {
             logger.error("Failed to invite participant: ", e);
+            // expire any channels that were successfully created, if the invite
+            // failed.
+            bridgeSession.colibriConference.expireChannels(colibriChannels);
```

Comment:
I think ParticipantChannel allocator has some logic about expiring channels if invite fails, it calls to `meetConference.onInviteFailed(this);` maybe it's a good time to have it unified? I think channel allocator is a better place to do it

Reply:
actuallyI'm not sure where to do, but it would be good to have it in one place that would cover both and regular participants

Reply:
We don't need this code, because `OctoChannelAllocator.invite()` is a no-op. 

I agree it would be nice to centralize error handling, but changing this code worries me because the side effects are so hard to follow. Perhaps the `ChannelAllocator` can check its status after `doRun` completes and clean up (expire channels) if necessary.

Reply:
You're right @paweldomas. This doesn't apply to Octo because there's no invite step, so I've removed the redundant code.

Reply:
@bgrozev I'm sorry I top-posted and resolved the conversation, I missed your latest message.

> Perhaps the ChannelAllocator can check its status after doRun completes and clean up (expire channels) if necessary.

I like this idea, I'll push a commit that implements this.